Library/Formula/bullet.rb

```diff
@@ -44,11 +44,20 @@ def install
 
     args << "-DUSE_DOUBLE_PRECISION=ON" if build.with? "double-precision"
 
-    args << "-DBUILD_DEMOS=OFF" if build.without? "demo"
+    with_demo = build.with? "demo"
+    if with_demo && build.with?("shared")
+      # Related to the following warnings when building --with-shared --with-demo
+      # https://gist.github.com/scpeters/6afc44f0cf916b11a226
+      opoo "Installed demos don't work with shared libraries"
+      puts <<-EOS.undent
+      Demos will be disabled due to the use of shared libraries.
+      Static libraries are required for demos installed by homebrew.
```

Comment:
* I increased the indentation in 398fac7621b5c4ed4787a84df6caa1f8a38c6145
* Let me clarify what I think the error message means:

The demos are not normally installed when one builds bullet, but in this formula there's nice `prefix.install` calls to install the appropriate build folders. This bring some extra cmake files, but that's not too big of a deal. The issue is that with shared libraries, the executables are linked using the absolute path of the build folder, such that after `prefix.install`ing, the libraries can no longer be found. There is also a [warning message](https://gist.github.com/scpeters/6afc44f0cf916b11a226) during install: `Warning: Could not fix ${lib} in ${executable}`.

I had added the `installed by homebrew` line in the warning message because trying to install them seems unique to homebrew.

That said, I'm open to changing the error message as you recommend. I just wanted to clarify why I had originally phrased it that way.

I'll wait for further comments and then update with the recommended message.